update docs/sources/git.md (#531)
operator-framework/rukpak: docs/sources/git.md

## Summary

A Bundle can reference content in a remote git repository instead of a remote container image by using the
`git` source type in the Bundle manifest. This enables one to easily source content from a git provider like GitHub and
make it available within the cluster.
The git source provides the contents in a git repository as the source of the bundle.  The `source.type` for the git source is `git`.
When creating a git source, a reference to a particular commit, tag, or branch must be provided in addition to the URL of the
repository and an optional directory in the repository. It is expected that a proper format of bundle content is present
in the particular commit/tag/branch at the directory specified.

For example, for a plain+v0 bundle, the git repository backing the bundle, or "bundle git repo", must have a certain
structure in order to produce a valid Bundle that works with the plain provisioner. It should have a manifests directory
where the Kubernetes manifests are stored -- this directory is used when copying manifests onto the cluster. By default,
the `manifests` directory is assumed to be at the root level, but a different bundle root can be specified

> via `spec.source.git.directory`.

Reviewer: Is it necessary to remove this description?

Author: Yes, this is about the `plain-v0` format.  It should go in `docs/bundles/plain.md`

Reviewer: Ah, right. Thank you.

> Referencing a git repository by a commit:

Reviewer: Can we make these be `###` tags to make the document flow a bit better? Applies to all subsequently similar lines.

```suggestion
### Referencing a git repository by a commit
```

spec:
  source:
    type: git
    git:
      ref:
        commit: d40082c96e6f0d297aa316d84020d307f95dc453
      repository: https://github.com/operator-framework/combo
  provisionerClassName: core-rukpak-io-plain
```

Referencing a git repository by a tag:

```yaml
apiVersion: core.rukpak.io/v1alpha1
kind: Bundle
metadata:
  name: combo-tag-ref
spec:
  source:
    type: git
    git:
      ref:
        tag: v0.0.2
      repository: https://github.com/operator-framework/combo
  provisionerClassName: core-rukpak-io-plain
```

Referencing a git repository by a branch:

```yaml
apiVersion: core.rukpak.io/v1alpha1
kind: Bundle
metadata:
  name: combo-branch-ref
spec:
  source:
    type: git
    git:
      ref:
        branch: main
      repository: https://github.com/operator-framework/combo
  provisionerClassName: core-rukpak-io-plain
```

> Referencing a different contents directory than the default:

Reviewer: ```suggestion
Referencing a different content directory than the default:
```

```yaml
apiVersion: core.rukpak.io/v1alpha1
kind: Bundle
metadata:
  name: combo-branch-ref-deploy
spec:
  source:
    type: git
    git:
      ref:
        branch: main
      directory: ./dev/deploy
      repository: https://github.com/exdx/combo-bundle
  provisionerClassName: core-rukpak-io-plain
```

## Private git repositories

> A git source can provide the contents in a private git repository by creating a secret in the namespace that the provisioner is deployed.

Reviewer: ```suggestion
A git source can reference content in a private git repository by creating a secret in the namespace that the provisioner is deployed.
```
